config.c

```diff
@@ -2316,10 +2316,23 @@ int git_config_get_virtualfilesystem(void)
 	if (core_virtualfilesystem && !*core_virtualfilesystem)
 		core_virtualfilesystem = NULL;
 
-	/* virtual file system relies on the sparse checkout logic so force it on */
 	if (core_virtualfilesystem) {
-		core_apply_sparse_checkout = 1;
-		return 1;
+		/*
+		 * Some git commands spawn helpers and redirect the index to a different
+		 * location.  These include "difftool -d" and the sequencer
+		 * (i.e. `git rebase -i`, `git cherry-pick` and `git revert`) and others.
+		 * In those instances we don't want to update their temporary index with
+		 * our virtualization data.
+		 */
+		char *default_index_file = xstrfmt("%s/%s", the_repository->gitdir, "index");
+		int should_run_hook = !strcmp(default_index_file, the_repository->index_file);
+
+		free(default_index_file);
+		if (should_run_hook) {
+			/* virtual file system relies on the sparse checkout logic so force it on */
+			core_apply_sparse_checkout = 1;
+			return 1;
+		}
```

Comment:
Do we need to turn off the `core_virtualfilesystem` flag in an else here or are all the checks of that flag inside a check of `core_apply_sparse_checkout`?  Because not it looks like we are in a state where the one is set but the other isn't.

Reply:
Good catch.  Yes, I'll update it to ensure they stay consistent.